### PR TITLE
Fix single precision interpolation in exporter and CLI

### PR DESCRIPTION
**Pull request by nilsvu**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: tests/Unit/IO/Exporter/Test_Exporter.cpp [hunk 1]
```diff
@@ -3,24 +3,80 @@
...
+    }
+    const auto interpolated_data = interpolate_to_points<2>(
+        h5_file_name, "/VolumeData", 0, {"Psi"}, {{{0.}, {0.}}});
+    // Compare to single precision
+    Approx custom_approx =
+        Approx::custom()
+            .epsilon(10. * std::numeric_limits<float>::epsilon())
+            .scale(1.0);
+    CHECK(interpolated_data[0][0] == custom_approx(1.));
+  }
```
Comment (nilsdeppe) [inline tests/Unit/IO/Exporter/Test_Exporter.cpp]: Can you delete the file here too? It's a bit annoying having the files hanging around IMO

File: tests/Unit/IO/Exporter/Test_Exporter.cpp [hunk 2]
```diff
@@ -3,24 +3,80 @@
...
+    const ElementId<2> element_id{0};
+    const Mesh<2> mesh{4, Spectral::Basis::Legendre,
+                       Spectral::Quadrature::GaussLobatto};
+    // Manufacture linear data so interpolation is exact
+    const auto xi = logical_coordinates(mesh);
+    const DataVector psi = 1.0 + get<0>(xi) + 2. * get<1>(xi);
+    // Convert to single precision
+    std::vector<float> psi_float(psi.size());
+    for (size_t i = 0; i < psi.size(); ++i) {
+      psi_float[i] = psi[i];
```
Comment (nilsdeppe) [inline tests/Unit/IO/Exporter/Test_Exporter.cpp]: `static_cast<float>(psi[i]);`

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
You can squash immediately

File: tests/Unit/IO/Exporter/Test_Exporter.cpp (hunk 1 above)
Comment (nilsvu) [reply tests/Unit/IO/Exporter/Test_Exporter.cpp]: Right, thanks for catching this

**Comment by nilsvu** [comment]
Done 👍 Thanks for reviewing!

**Review by nilsdeppe — DISMISSED** [review]

**Comment by nilsvu** [comment]
I messed up curly brackets, fixed now

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**